### PR TITLE
tests: add ubuntu 20.10 to spread tests

### PR DESCRIPTION
**Pull request by sergiocazzolato**

### CODE REVIEW

**Comment by zyga** [comment]
The preseed tasks are failing on 20.10:
```
+ systemd-run --system --service-type=forking --unit=qemu-ndb-preseed.service '' --fork -c /dev/nbd0 /home/gopath/src/github.com/snapcore/snapd/tests/main/preseed/cloudimg.img
Job for qemu-ndb-preseed.service failed because the control process exited with error code.
See "systemctl status qemu-ndb-preseed.service" and "journalctl -xe" for details.
```
CC @stolowski

**Comment by stolowski** [comment]
> The preseed tasks are failing on 20.10:
> 
> ```
> + systemd-run --system --service-type=forking --unit=qemu-ndb-preseed.service '' --fork -c /dev/nbd0 /home/gopath/src/github.com/snapcore/snapd/tests/main/preseed/cloudimg.img
> Job for qemu-ndb-preseed.service failed because the control process exited with error code.
> See "systemctl status qemu-ndb-preseed.service" and "journalctl -xe" for details.
> ```
> 
> CC @stolowski

That's because qemu-nbd command is not found, I think pkgdb.sh needs to be updated with 20.10 case.

**Comment by sergiocazzolato** [comment]
@stolowski thanks for the review. Comman qemu-ndb not included in qemu-utils package. I'll skip those tests until it is fixed upstream. 

google:ubuntu-20.10-64 .../tests/main/preseed# command  qemu-ndb Command 'qemu-ndb' not found, did you mean:   command 'qemu-nbd' from deb qemu-utils (1:4.2-3ubuntu8) Try: apt install <deb name>  google:ubuntu-20.10-64 .../tests/main/preseed# sudo apt install qemu-utils Reading package lists... Done Building dependency tree        Reading state information... Done qemu-utils is already the newest version (1:4.2-3ubuntu8). 0 upgraded, 0 newly installed, 0 to remove and 13 not upgraded.  google:ubuntu-20.10-64 .../tests/main/preseed# find / -name qemu-ndb  google:ubuntu-20.10-64 .../tests/main/preseed# apt-cache policy qemu-utils qemu-utils:   Installed: 1:4.2-3ubuntu8   Candidate: 1:4.2-3ubuntu8   Version table:  *** 1:4.2-3ubuntu8 500         500 http://us-east1.gce.archive.ubuntu.com/ubuntu groovy/main amd64 Packages         100 /var/lib/dpkg/status
--

**Comment by sergiocazzolato** [comment]
@stolowski I included a fix for typo in qemu-ndb as well

File: tests/main/preseed-reset/task.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ description: |
   This test checks that preseeding of Ubuntu cloud images with snap-preseed
   can be undone with --reset flag.
 
-systems: [ubuntu-19.10-*, ubuntu-2*]
+systems: [ubuntu-19.10-*, ubuntu-20*]
```
Comment (mvo5) [inline tests/main/preseed-reset/task.yaml:6]: I think we want a TODO here as well, I guess the same/similar to the one for lxd?

File: tests/main/preseed/task.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ description: |
   command works, up to the point where the image is ready to be booted.
   The test assumes cloud image with a core and lxd snaps in its seeds/.
 
-systems: [ubuntu-19.10-*, ubuntu-2*]
```
Comment (mvo5) [inline tests/main/preseed/task.yaml:7]: And a coment here too?

File: tests/main/preseed-lxd/task.yaml [hunk 3]
```diff
@@ -18,9 +18,9 @@ prepare: |
 
   modprobe nbd
 
-  # Run qemu-ndb as a service, so that it does not interact with ssh
+  # Run qemu-nbd as a service, so that it does not interact with ssh
   # stdin/stdout it would otherwise inherit from the spread session.
-  systemd-run --system --service-type=forking --unit=qemu-ndb-preseed.service "$(command -v qemu-nbd)" --fork -c /dev/nbd0 "$(pwd)/cloudimg.img"
+  systemd-run --system --service-type=forking --unit=qemu-nbd-preseed.service "$(command -v qemu-nbd)" --fork -c /dev/nbd0 "$(pwd)/cloudimg.img"
```
Comment (stolowski) [inline tests/main/preseed-lxd/task.yaml:23]: Thanks for fixing the typos!

File: tests/main/preseed-reset/task.yaml (hunk 1 above)
Comment (stolowski) [reply tests/main/preseed-reset/task.yaml:6]: My understanding (after calrifying qemu-utils confusion on 20.10) was that these tests pass on 20.10? If not, then I agree, please add a comment explaining why 20.10 is excluded.

**Comment by sergiocazzolato** [comment]
@mvo5 thanks for the review, I already update the filters on the tests to fix that issue.

File: tests/main/preseed-lxd/task.yaml (hunk 3 above)
Comment (sergiocazzolato) [reply tests/main/preseed-lxd/task.yaml:23]: yaw

**Pull request merged**